core/common/src/Clock.kt

```diff
@@ -11,15 +11,41 @@ import kotlin.time.*
  * A source of [Instant] values.
  *
  * See [Clock.System][Clock.System] for the clock instance that queries the operating system.
+ *
+ * It is recommended not to use [Clock.System] directly in the implementation; instead, one could pass a
+ * [Clock] explicitly to the functions or classes that need it.
+ * This way, tests can be written deterministically by providing custom [Clock] implementations
+ * to the system under test.
  */
 public interface Clock {
     /**
      * Returns the [Instant] corresponding to the current time, according to this clock.
+     *
+     * It is not guaranteed that calling [now] later will return a larger [Instant].
+     * In particular, for [System], violations of this are completely expected and must be taken into account.
```

Comment:
Maybe instead of "violations of this", it is better to phrase it as 

> In particular, for System this behavior is expected and...

Reply:
Better to reference `System` as `Clock.System`